### PR TITLE
Added an algorithm that approximates line lengths

### PR DESCRIPTION
**Pull request by shrabian**
### **Describe your change:**
Added an algorithm that approximates the length of a line segment whose curve is defined by a function. This relates to issue #1688 

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.

### CODE REVIEW

File: maths/line_length.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+from typing import Callable, Union
+import math as m
+
+def line_length(fnc: Callable[[Union[int, float]], Union[int, float]],
+                x_start: Union[int, float],
+                x_end: Union[int, float],
+                steps: int = 100) -> float:
```
Comment (cclauss) [inline maths/line_length.py]: This is returning __str__, not __float__.

File: maths/line_length.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    :return: a float representing the length of the curve
+
+    >>> def f(x):
+    ...    return x
+    >>> '%.6f' % line_length(f, 0, 1, 10)
+    '1.414214'
+
+    >>> def f(x):
+    ...    return 1
+    >>> '%.6f' % line_length(f, -5.5, 4.5, 100)
```
Comment (cclauss) [inline maths/line_length.py]: Please remove the trailing 100 parameter in this test to demonstrate that there is default value and it works.

File: maths/line_length.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    sequence of linear lines and summing their lengths
+    :param fnc: a function which defines a curve
+    :param x_start: left end point to indicate the start of line segment
+    :param x_end: right end point to indicate end of line segment
+    :param steps: an accuracy gauge; more steps increases accuracy
+    :return: a float representing the length of the curve
+
+    >>> def f(x):
+    ...    return x
+    >>> '%.6f' % line_length(f, 0, 1, 10)
```
Comment (cclauss) [inline maths/line_length.py]: [f-strings](https://realpython.com/python-f-strings) are shorter and faster...
```
f"{line_length(f, 0, 1, 10):.6f}"
```

File: maths/line_length.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    def f(x):
+        return m.sin(10*x)
+
+    print("f(x) = sin(10 * x)")
+    print("The length of the curve from x = -10 to x = 10 is:")
+    i = 10
+    while i <= 100000:
+        length = line_length(f, -10, 10, i)
+        print("With {} steps: {}".format(i, length))
```
Comment (cclauss) [inline maths/line_length.py]:
```python
print(f"With {i} steps: {line_length(f, -10, 10, i)}")
```

File: maths/line_length.py (hunk 1 above)
Comment (shrabian) [reply maths/line_length.py]: My function should return a float. In the doctests, I've converted the return value into a string to allow formatting. I've kept it as float for now, let me know if that's ok.

**Pull request merged**